### DISCUSSION TITLE
airflow standalone triggers ModuleNotFoundError: No module named '_sqlite3'

### DISCUSSION THREAD

**Discussion by matheuscirillo** [apache/airflow#30470] [Q&A] [answered]
So I already read about 20 topics on this subjetct, but I've haven't found a solution yet.

OS: Ubuntu 22.04

I'm using the python installation that came with Ubuntu.
```
> python3 --version
> Python 3.10.4
```

I just installed airflow through pip and when I run airflow standalone, the error `ModuleNotFoundError: No module named '_sqlite3'` shows up.

I thought I didn't had sqlite3 package installed so I ran:

```
> pip3 install sqlite3
```

The output was: 
```
> Defaulting to user installation because normal site-packages is not writeable
> ERROR: Could not find a version that satisfies the requirement sqlite3 (from versions: none)
> ERROR: No matching distribution found for sqlite3
```

Ok, I found that I should do pip3 install pysqlite3 instead.
Did so:

```
> pip3 install pysqlite3                
> Defaulting to user installation because normal site-packages is not writeable
> Requirement already satisfied: pysqlite3 in ./.local/lib/python3.10/site-packages (0.5.0)
```

Already tried a bunch of solutions that I read.

1. Install sqlite library on Ubuntu through apt-get:
```
sudo apt-get install libsqlite3-dev
```

2. Installed db-sqlite3 through pip3
```
pip3 install db-sqlite3
```

I event uninstalled and installed all these things, but nothing seems to work.
Any solutions?

**Comment by matheuscirillo** [comment, marked as answer]
So I had to rebuild Python and install.

I followed the following:

Check if the '_sqlite3' module is available in your Python installation:

```
> python3 -c "import _sqlite3"
```

The ModuleNotFoundError pops up.

I guessed that, my python installation was broken someway, then I ran the following commands:

1. Install the necessary build dependencies:

```
> sudo apt-get install build-essential libssl-dev zlib1g-dev libncurses5-dev libncursesw5-dev libreadline-dev libsqlite3-dev libgdbm-dev libdb5.3-dev libbz2-dev libexpat1-dev liblzma-dev tk-dev libffi-dev
```

2. Download the latest version of Python from the official website:

```
> wget https://www.python.org/ftp/python/3.10.4/Python-3.10.4.tgz
```

3. Extract the downloaded archive:

```
> tar -xvf Python-3.10.4.tgz
```

4. Configure the build with the necessary options:

```
> cd Python-3.10.4
> ./configure --enable-optimizations --with-ensurepip=install
```

5. Build and install Python

```
> make -j $(nproc)
> sudo make altinstall
```

Now it's working as expected ;)

**Reply by jack-b-kelly** [reply]
Thank you!

**Reply by Daremitsu1** [reply]
Works for Python 3.7. Fantastic!

**Reply by Olimar1164** [reply]
Works for Python 3.11.7, Thanks!

**Reply by MMnaimi** [reply]
Works for python 3.10.13, Thanks!

**Reply by mashal02** [reply]
Works for Python 3.12.2, thank you! Got a few errors in between and chatgpt'd them, and got them fixed. Otherwise works and solved my issue!

**Reply by ken19980727** [reply]
Works for python 3.9.12, Thanks!

**Reply by Kelverton** [reply]
Man, really thanks, it worked so well in python3.12.3 ! Many thanks !

**Reply by kubada** [reply]
Works for python 3.12.5, thx!

**Reply by TiemDang** [reply]
You save me. Thank you. It work for python 3.10.12.

**Comment by codingboyuz** [comment]
Works for python 3.12.6, Thanks!
